modules/http/http.c

```diff
@@ -282,6 +299,68 @@ http_dd_get_template_options(LogDriver *d)
   return &self->template_options;
 }
 
+void
+http_dd_set_ca_dir(LogDriver *d, const gchar *ca_dir)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  g_free(self->ca_dir);
+  self->ca_dir = g_strdup(ca_dir);
+}
+
+void
+http_dd_set_ca_file(LogDriver *d, const gchar *ca_file)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  g_free(self->ca_file);
+  self->ca_file = g_strdup(ca_file);
+}
+
+void
+http_dd_set_cert_file(LogDriver *d, const gchar *cert_file)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  g_free(self->cert_file);
+  self->cert_file = g_strdup(cert_file);
+}
+
+void
+http_dd_set_key_file(LogDriver *d, const gchar *key_file)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  g_free(self->key_file);
+  self->key_file = g_strdup(key_file);
+}
+
+void
+http_dd_set_cipher_suite(LogDriver *d, const gchar *ciphers)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  g_free(self->ciphers);
+  self->ciphers = g_strdup(ciphers);
+}
+
+void
+http_dd_set_peer_verify(LogDriver *d, const gchar *value)
+{
+  gboolean verify;
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) d;
+
+  if (type_cast_to_boolean(value, &verify, NULL))
+    {
+      self->peer_verify = verify;
+    }
+  else
+    {
+      msg_warning("curl: non-boolean value for TLS peer verify",
+                  evt_tag_str("peer_verify", value));
+    }
+}
```

Comment:
Apart from this function this patch set looks great. There are two issues from which one might only be a question:

1) is peer verify only a binary switch? Iirc the to destination allow a set of options there and we should keep this consistent

2) even if peer verify should be a boolean the tour safety functions are intended for a very different use case (value pairs type casting). So this function shouldn't be used here. Boolean options can be parsed using the yesno rule in the grammar, which yields an integer that can then be used to set this boolean value directly.